### ISSUE TITLE
Pre-trained tokenizer `repr` is inconsistent with attribute name

### ISSUE THREAD

**Issue by inwaves** [huggingface/transformers#21073]
### System Info

- `transformers` version: 4.26.0.dev0
- Platform: macOS-12.5.1-x86_64-i386-64bit
- Python version: 3.10.8
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 1.13.1 (False)
- Tensorflow version (GPU?): 2.11.0 (False)
- Flax version (CPU?/GPU?/TPU?): 0.6.3 (cpu)
- Jax version: 0.4.1
- JaxLib version: 0.4.1
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

Hi @ArthurZucker, since this is tokeniser-related, do you mind having a look?

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

This one's pretty straightforward: 

1. Using a pre-trained tokeniser (`PreTrainedTokenizerFast` or `PreTrainedTokenizerBase`), print out the object.
2. The `repr`, which is defined in [`tokenization_utils_base`](https://github.com/huggingface/transformers/blob/a3c37825cc1e305dde63455b5f321586e6d29e07/src/transformers/tokenization_utils_base.py#L1573), returns something like this:

`PreTrainedTokenizerFast(name_or_path='gpt2', vocab_size=50257, model_max_len=1024, is_fast=True, padding_side='right', truncation_side='right', special_tokens={'bos_token': '<|endoftext|>', 'eos_token': '<|endoftext|>', 'unk_token': '<|endoftext|>', 'pad_token': '[PAD]'})`

3. Note the `model_max_len` attribute.

### Expected behavior

The repr should display `model_max_length=1024` instead, since that is the actual name of the attribute. Other attribute labels in the repr seem consistent with the name, which leads me to believe this is a typo.

I came across this because I printed out the object, and then tried to access that tokeniser's `model_max_len`, which of course errors out since there's no attribute with that name.

**Comment by ArthurZucker** [comment]
The `PreTrainedTokenizerFast` and `PreTrainedTokenizerBase` are abstract classes and should not really be used. The `model_max_len` is a vestige of a previous argument, opening a PR to fix this typo as indeed the attribute does not exist.
